{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# please don't use this search tool for more than 2 times per day.\n",
    "## (Google have a limitation on code accessment, otherwise it will block our future access.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Start from here\n",
    "## Change keyword inside bracket []\n",
    "## such as [\"Treg scRNAseq\"] with the content you need to search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "keyword = [\"CRISPR analysis\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading next 10 results\n",
      "./src/sortgs.py:253: UserWarning: Year not found for Could not catch title, appending 0\n",
      "  warnings.warn(\"Year not found for {}, appending 0\".format(title[-1]))\n",
      "Loading next 20 results\n",
      "Loading next 30 results\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Author</th>\n",
       "      <th>Title</th>\n",
       "      <th>Citations</th>\n",
       "      <th>Year</th>\n",
       "      <th>Source</th>\n",
       "      <th>cit/year</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rank on Google Scholar</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cho, S Kim, Y Kim, J Kweon, HS Kim…</td>\n",
       "      <td>Analysis of off-target effects of CRISPR/Cas-d...</td>\n",
       "      <td>1512</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://genome.cshlp.org/content/24/1/132.short</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Feng, Y Mao, N Xu, B Zhang, P Wei…</td>\n",
       "      <td>Multigeneration analysis reveals the inheritan...</td>\n",
       "      <td>685</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://www.pnas.org/doi/abs/10.1073/pnas.1400...</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Peters, A Colavin, H Shi, TL Czarny, MH Larson…</td>\n",
       "      <td>A comprehensive, CRISPR-based functional analy...</td>\n",
       "      <td>478</td>\n",
       "      <td>2016</td>\n",
       "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Hsiau, D Conant, N Rossi, T Maures, K Waite, J...</td>\n",
       "      <td>Inference of CRISPR edits from Sanger trace data</td>\n",
       "      <td>252</td>\n",
       "      <td>2019</td>\n",
       "      <td>https://www.biorxiv.org/content/10.1101/251082...</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Anderson, M Haeussler, C Watanabe…</td>\n",
       "      <td>CRISPR off-target analysis in genetically engi...</td>\n",
       "      <td>158</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://www.nature.com/articles/s41592-018-0011-5</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Wang, M Wang, W Zhang, T Xiao, CH Chen, A Wu…</td>\n",
       "      <td>Integrative analysis of pooled CRISPR genetic ...</td>\n",
       "      <td>116</td>\n",
       "      <td>2019</td>\n",
       "      <td>https://www.nature.com/articles/s41596-018-0113-7</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Hanna, JG Doench</td>\n",
       "      <td>Design and analysis of CRISPR–Cas experiments</td>\n",
       "      <td>85</td>\n",
       "      <td>2020</td>\n",
       "      <td>https://www.nature.com/articles/s41587-020-0490-7</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Horvath, AC Coût</td>\n",
       "      <td>Comparative analysis of CRISPR loci in lactic ...</td>\n",
       "      <td>328</td>\n",
       "      <td>2009</td>\n",
       "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Shmakov, KS Makarova, YI Wolf…</td>\n",
       "      <td>Systematic prediction of genes functionally li...</td>\n",
       "      <td>99</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://www.pnas.org/doi/abs/10.1073/pnas.1803...</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Hwang, J Park, K Lim, S Kim, J Yu, E Yu, ST Kim…</td>\n",
       "      <td>Web-based design and analysis tools for CRISPR...</td>\n",
       "      <td>94</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://link.springer.com/article/10.1186/s128...</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Connelly, SM Pruet</td>\n",
       "      <td>CRIS. py: a versatile and high-throughput anal...</td>\n",
       "      <td>70</td>\n",
       "      <td>2019</td>\n",
       "      <td>https://www.nature.com/articles/s41598-019-408...</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Leto, DW Morgens, L Zhang, CP Walczak, JE Elias…</td>\n",
       "      <td>Genome-wide CRISPR analysis identifies substra...</td>\n",
       "      <td>62</td>\n",
       "      <td>2019</td>\n",
       "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Crawley, JR Henriksen, R Barrangou</td>\n",
       "      <td>CRISPRdisco: an automated pipeline for the dis...</td>\n",
       "      <td>69</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://www.liebertpub.com/doi/abs/10.1089/cri...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Sheng, P Wang, J Yang, L Tang, F Chen…</td>\n",
       "      <td>MXene Coupled with CRISPR-Cas12a for analysis ...</td>\n",
       "      <td>26</td>\n",
       "      <td>2021</td>\n",
       "      <td>https://pubs.acs.org/doi/abs/10.1021/acs.analc...</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>van der Ploeg</td>\n",
       "      <td>Analysis of CRISPR in Streptococcus mutans sug...</td>\n",
       "      <td>173</td>\n",
       "      <td>2009</td>\n",
       "      <td>https://www.microbiologyresearch.org/content/j...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Savitskaya, E Semenova, V Dedkov, A Metlitskaya…</td>\n",
       "      <td>High-throughput analysis of type IE CRISPR/Cas...</td>\n",
       "      <td>118</td>\n",
       "      <td>2013</td>\n",
       "      <td>https://www.tandfonline.com/doi/abs/10.4161/rn...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Labun, X Guo, A Chavez, G Church…</td>\n",
       "      <td>Accurate analysis of genuine CRISPR editing ev...</td>\n",
       "      <td>35</td>\n",
       "      <td>2019</td>\n",
       "      <td>https://genome.cshlp.org/content/29/5/843.short</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Bodapati, TP Daley, X Lin…</td>\n",
       "      <td>A benchmark of algorithms for the analysis of ...</td>\n",
       "      <td>28</td>\n",
       "      <td>2020</td>\n",
       "      <td>https://genomebiology.biomedcentral.com/articl...</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>You, Z Zhong, Q Ren, F Hassan…</td>\n",
       "      <td>CRISPRMatch: an automatic calculation and visu...</td>\n",
       "      <td>38</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Hwang, YK Jeong, O Habib, SA Hong…</td>\n",
       "      <td>PE-Designer and PE-Analyzer: web-based design ...</td>\n",
       "      <td>17</td>\n",
       "      <td>2021</td>\n",
       "      <td>https://academic.oup.com/nar/article-abstract/...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Waldrip, SD Byrum, AJ Storey, J Gao, AK Byrd…</td>\n",
       "      <td>A CRISPR-based approach for proteomic analysis...</td>\n",
       "      <td>75</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://www.tandfonline.com/doi/abs/10.4161/ep...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Chen, T Xiao, H Xu, P Jiang, CA Meyer, W Li…</td>\n",
       "      <td>Improved design and analysis of CRISPR knockou...</td>\n",
       "      <td>33</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://academic.oup.com/bioinformatics/articl...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Wang, C Tilford, I Neuhaus, G Mintier, Q Guo…</td>\n",
       "      <td>CRISPR-DAV: CRISPR NGS data analysis and visua...</td>\n",
       "      <td>41</td>\n",
       "      <td>2017</td>\n",
       "      <td>https://academic.oup.com/bioinformatics/articl...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Semenova, M Nagornykh, M Pyatnitskiy…</td>\n",
       "      <td>Analysis of CRISPR system function in plant pa...</td>\n",
       "      <td>92</td>\n",
       "      <td>2009</td>\n",
       "      <td>https://academic.oup.com/femsle/article-abstra...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Mills, C Griffin, A Coffey, WC Meijer…</td>\n",
       "      <td>CRISPR analysis of bacteriophage‐insensitive m...</td>\n",
       "      <td>78</td>\n",
       "      <td>2010</td>\n",
       "      <td>https://sfamjournals.onlinelibrary.wiley.com/d...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   Author  \\\n",
       "Rank on Google Scholar                                                      \n",
       "3                                     Cho, S Kim, Y Kim, J Kweon, HS Kim…   \n",
       "15                                     Feng, Y Mao, N Xu, B Zhang, P Wei…   \n",
       "4                         Peters, A Colavin, H Shi, TL Czarny, MH Larson…   \n",
       "18                      Hsiau, D Conant, N Rossi, T Maures, K Waite, J...   \n",
       "14                                     Anderson, M Haeussler, C Watanabe…   \n",
       "13                          Wang, M Wang, W Zhang, T Xiao, CH Chen, A Wu…   \n",
       "1                                                        Hanna, JG Doench   \n",
       "9                                                        Horvath, AC Coût   \n",
       "29                                         Shmakov, KS Makarova, YI Wolf…   \n",
       "17                       Hwang, J Park, K Lim, S Kim, J Yu, E Yu, ST Kim…   \n",
       "27                                                     Connelly, SM Pruet   \n",
       "24                       Leto, DW Morgens, L Zhang, CP Walczak, JE Elias…   \n",
       "16                                     Crawley, JR Henriksen, R Barrangou   \n",
       "26                                 Sheng, P Wang, J Yang, L Tang, F Chen…   \n",
       "7                                                           van der Ploeg   \n",
       "20                       Savitskaya, E Semenova, V Dedkov, A Metlitskaya…   \n",
       "8                                       Labun, X Guo, A Chavez, G Church…   \n",
       "6                                              Bodapati, TP Daley, X Lin…   \n",
       "33                                         You, Z Zhong, Q Ren, F Hassan…   \n",
       "28                                     Hwang, YK Jeong, O Habib, SA Hong…   \n",
       "19                          Waldrip, SD Byrum, AJ Storey, J Gao, AK Byrd…   \n",
       "21                           Chen, T Xiao, H Xu, P Jiang, CA Meyer, W Li…   \n",
       "11                          Wang, C Tilford, I Neuhaus, G Mintier, Q Guo…   \n",
       "10                                  Semenova, M Nagornykh, M Pyatnitskiy…   \n",
       "32                                 Mills, C Griffin, A Coffey, WC Meijer…   \n",
       "\n",
       "                                                                    Title  \\\n",
       "Rank on Google Scholar                                                      \n",
       "3                       Analysis of off-target effects of CRISPR/Cas-d...   \n",
       "15                      Multigeneration analysis reveals the inheritan...   \n",
       "4                       A comprehensive, CRISPR-based functional analy...   \n",
       "18                       Inference of CRISPR edits from Sanger trace data   \n",
       "14                      CRISPR off-target analysis in genetically engi...   \n",
       "13                      Integrative analysis of pooled CRISPR genetic ...   \n",
       "1                           Design and analysis of CRISPR–Cas experiments   \n",
       "9                       Comparative analysis of CRISPR loci in lactic ...   \n",
       "29                      Systematic prediction of genes functionally li...   \n",
       "17                      Web-based design and analysis tools for CRISPR...   \n",
       "27                      CRIS. py: a versatile and high-throughput anal...   \n",
       "24                      Genome-wide CRISPR analysis identifies substra...   \n",
       "16                      CRISPRdisco: an automated pipeline for the dis...   \n",
       "26                      MXene Coupled with CRISPR-Cas12a for analysis ...   \n",
       "7                       Analysis of CRISPR in Streptococcus mutans sug...   \n",
       "20                      High-throughput analysis of type IE CRISPR/Cas...   \n",
       "8                       Accurate analysis of genuine CRISPR editing ev...   \n",
       "6                       A benchmark of algorithms for the analysis of ...   \n",
       "33                      CRISPRMatch: an automatic calculation and visu...   \n",
       "28                      PE-Designer and PE-Analyzer: web-based design ...   \n",
       "19                      A CRISPR-based approach for proteomic analysis...   \n",
       "21                      Improved design and analysis of CRISPR knockou...   \n",
       "11                      CRISPR-DAV: CRISPR NGS data analysis and visua...   \n",
       "10                      Analysis of CRISPR system function in plant pa...   \n",
       "32                      CRISPR analysis of bacteriophage‐insensitive m...   \n",
       "\n",
       "                        Citations  Year  \\\n",
       "Rank on Google Scholar                    \n",
       "3                            1512  2014   \n",
       "15                            685  2014   \n",
       "4                             478  2016   \n",
       "18                            252  2019   \n",
       "14                            158  2018   \n",
       "13                            116  2019   \n",
       "1                              85  2020   \n",
       "9                             328  2009   \n",
       "29                             99  2018   \n",
       "17                             94  2018   \n",
       "27                             70  2019   \n",
       "24                             62  2019   \n",
       "16                             69  2018   \n",
       "26                             26  2021   \n",
       "7                             173  2009   \n",
       "20                            118  2013   \n",
       "8                              35  2019   \n",
       "6                              28  2020   \n",
       "33                             38  2018   \n",
       "28                             17  2021   \n",
       "19                             75  2014   \n",
       "21                             33  2018   \n",
       "11                             41  2017   \n",
       "10                             92  2009   \n",
       "32                             78  2010   \n",
       "\n",
       "                                                                   Source  \\\n",
       "Rank on Google Scholar                                                      \n",
       "3                         https://genome.cshlp.org/content/24/1/132.short   \n",
       "15                      https://www.pnas.org/doi/abs/10.1073/pnas.1400...   \n",
       "4                       https://www.sciencedirect.com/science/article/...   \n",
       "18                      https://www.biorxiv.org/content/10.1101/251082...   \n",
       "14                      https://www.nature.com/articles/s41592-018-0011-5   \n",
       "13                      https://www.nature.com/articles/s41596-018-0113-7   \n",
       "1                       https://www.nature.com/articles/s41587-020-0490-7   \n",
       "9                       https://www.sciencedirect.com/science/article/...   \n",
       "29                      https://www.pnas.org/doi/abs/10.1073/pnas.1803...   \n",
       "17                      https://link.springer.com/article/10.1186/s128...   \n",
       "27                      https://www.nature.com/articles/s41598-019-408...   \n",
       "24                      https://www.sciencedirect.com/science/article/...   \n",
       "16                      https://www.liebertpub.com/doi/abs/10.1089/cri...   \n",
       "26                      https://pubs.acs.org/doi/abs/10.1021/acs.analc...   \n",
       "7                       https://www.microbiologyresearch.org/content/j...   \n",
       "20                      https://www.tandfonline.com/doi/abs/10.4161/rn...   \n",
       "8                         https://genome.cshlp.org/content/29/5/843.short   \n",
       "6                       https://genomebiology.biomedcentral.com/articl...   \n",
       "33                      https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...   \n",
       "28                      https://academic.oup.com/nar/article-abstract/...   \n",
       "19                      https://www.tandfonline.com/doi/abs/10.4161/ep...   \n",
       "21                      https://academic.oup.com/bioinformatics/articl...   \n",
       "11                      https://academic.oup.com/bioinformatics/articl...   \n",
       "10                      https://academic.oup.com/femsle/article-abstra...   \n",
       "32                      https://sfamjournals.onlinelibrary.wiley.com/d...   \n",
       "\n",
       "                        cit/year  \n",
       "Rank on Google Scholar            \n",
       "3                            168  \n",
       "15                            76  \n",
       "4                             68  \n",
       "18                            63  \n",
       "14                            32  \n",
       "13                            29  \n",
       "1                             28  \n",
       "9                             23  \n",
       "29                            20  \n",
       "17                            19  \n",
       "27                            18  \n",
       "24                            16  \n",
       "16                            14  \n",
       "26                            13  \n",
       "7                             12  \n",
       "20                            12  \n",
       "8                              9  \n",
       "6                              9  \n",
       "33                             8  \n",
       "28                             8  \n",
       "19                             8  \n",
       "21                             7  \n",
       "11                             7  \n",
       "10                             7  \n",
       "32                             6  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!python ./src/sortgs.py --kw $keyword --sortby \"cit/year\"\n",
    "df = pd.read_csv(\"[\"+ keyword[0].replace(\" \", \"_\") + \"].csv\", index_col=\"Rank on Google Scholar\")\n",
    "df.head(25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
